src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -246,6 +257,37 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
             }
         }
 
+        private bool TryNavigateToMappedFile(Workspace workspace, ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            Contract.ThrowIfNull(vsWorkspace);
```

Comment:
In this case hard cast, so that way you'll get an InvalidCastException which will show the other underlying type if it ever fails.